packages/rum/src/apm-base.js

```diff
@@ -74,10 +74,46 @@ class ApmBase {
       if (flags[PAGE_LOAD] && configService.get('sendPageLoadTransaction')) {
         this._sendPageLoadMetrics()
       }
+
+      if (configService.get('centralConfig')) {
+        this.fetchCentralConfig()
+      }
     }
     return this
   }
 
+  fetchCentralConfig() {
+    const apmServer = this.serviceFactory.getService('ApmServer')
+    const loggingService = this.serviceFactory.getService('LoggingService')
+    const configService = this.serviceFactory.getService('ConfigService')
+
+    return apmServer
+      .fetchConfig(
+        configService.get('serviceName'),
+        configService.get('environment')
+      )
+      .then(config => {
+        var transactionSampleRate = config['transaction_sample_rate']
+        if (transactionSampleRate) {
+          transactionSampleRate = Number(transactionSampleRate)
+          const config = { transactionSampleRate }
```

Comment:
this whole logic can be replaced with `this.config(config)` and the function will take care of the logging the error message and setting config if its valid.

Reply:
Since the message here says value, we can replace this generic error logic with `value` instead of `character`. https://github.com/elastic/apm-agent-rum-js/blob/c38f175a3d9b8352c2d153318eb2ccf5c67e91c7/packages/rum/src/apm-base.js#L155

Reply:
`this.config` checks for `serviceName` and `serverUrl` which are required but we don't set them from the remote config.